### PR TITLE
reverts the cult stun change and nerfs barrier/final reckoning instead

### PR DESCRIPTION
**Pull request by imsxz**
## About The Pull Request
cult code gave me a headache, give me an @ if you see something you dont like. this PR changes a lot of cult code

## Why It's Good For The Game
the cult stun nerf was really dumb and hamfisted by someone who didnt realize how cult actually wins after gaining a small foothold. with their nerf, cult is forced to avoid fighting mindshielded people at all costs(of which they cant even identify until AFTER they try to stun). The cult stun is balanced around the fact that cult is severely weak to ranged ballistics (read: the shotgun) and keeps cult able to take on security in melee. The previous nerf made cult weaker on BOTH fronts.

I made sure to play 2 rounds of cult before making this change, one as cult and one as non-cult. Not a large sample size but better than nothing. I was able to fight cult without even stunning them prior by just hitting them with a baseball bat, didnt even matter that i was an assistant with no other gear when the stunned me because i had gotten a mindshield from a crate outside of cargo. It was really dumb, and it felt like I cheated him out of a cult round.

Earlier, I had played as a cultist. After converting several non-implanted people and sacrificing our mindshielded target that a changeling had killed prior, I barrier'd myself into a room, scribed the narsie rune, and then used final reckoning as a horde of angry crewmembers lay staring dumbfounded at my omnipotent barrier rune. We straight up would have gotten overwhelmed if they could force past the barriers, as we were 100% outnumbered and outgunned. 

This PR reverts the stun change for the above mentioned reasons, and introduces 2 more severe nerfs that I argue will have a bigger effect on win rate while also making cult summon feel a LOT more fair by giving the crew a chance  to stop them.

Specifically, barrier rune can now be destroyed. It has 100 integrity, and erases the rune below it when it is broken. To prevent abuse, barrier runes cant be constantly invoked to replenish them, the integrity is persistent after lowering it. Clicking the barrier itself as a cult construct or cultist will invoke the rune to allow passage. Simple mobs that arent cultists are still able to smash the barrier.

Final reckoning has been limited to only areas that narsie cant be summoned in. This is to prevent the cheese/instant summons of the past which have often killed the mood as it removes the fight out of the final summon, usually combined with barrier runes. I have done this myself dozens of times. I did it earlier today even with the stun nerf to win. Masters will still be able to summon their cult outside of the summon location prior to get everyone near the location to prepare defending.

tl;dr removes the stun nerf, makes barrier runes breakable, cant final reckoning in summon locations

## Changelog
:cl:imsxz
balance: reverts the recent cult stun nerf, allowing mindshielded people to be stunned again
balance: cult barrier runes can be smashed until they break
balance: cult leaders can no longer use final reckoning in narsie summon locations
/:cl:

### CODE REVIEW

**Comment by Archanial** [comment]
> avoid fighting mindshielded people at all costs(of which they cant even identify until AFTER they try to stun)

If not seeing who's mindshielded later in game is that big of a problem add sechuds to blindfolds.
> The cult stun is balanced around the fact that cult is severely weak to ranged ballistics 

How on earth is one-hit stun balanced? Have you tried playing security fighting cultists? Most of the time random security officers won't have shotguns ready shift-start to blast you with and using stun you still can try to steal baton from them and win the fight that way.

**Comment by ArcaneDefence** [comment]
> Have you tried playing security fighting cultists?

>I was able to fight cult without even stunning them prior by just hitting them with a baseball bat, didnt even matter that i was an assistant with no other gear when the stunned me because i had gotten a mindshield from a crate outside of cargo.

Definitely seems like there's some room for discussion if you can have people going around with baseball bats made out of the curator's desk with the mindshields from the armory just bludgeoning down the bloodcult, when they are supposedly strong in melee.

**Comment by wesoda25** [comment]
bye bye solo cult!

**Comment by imsxz** [comment]
> 
> 
> > Have you tried playing security fighting cultists?
> 
> > I was able to fight cult without even stunning them prior by just hitting them with a baseball bat, didnt even matter that i was an assistant with no other gear when the stunned me because i had gotten a mindshield from a crate outside of cargo.
> 
> Definitely seems like there's some room for discussion if you can have people going around with baseball bats made out of the curator's desk with the mindshields from the armory just bludgeoning down the bloodcult, when they are supposedly strong in melee.

they arent strong in melee with the change is my point.

**Comment by imsxz** [comment]
> 
> 
> bye bye solo cult!

team gamemode

**Comment by Shadowflame909** [comment]
Some Cult Gear has negative armor (Flagellant Robes) So all it really takes is one flash bang and a shotgun to genocide an entire cult.

Pretty sure the Shotgun coder is gonna be nerfing that eventually though.

**Comment by imsxz** [comment]
> 
> 
> > avoid fighting mindshielded people at all costs(of which they cant even identify until AFTER they try to stun)
> 
> If not seeing who's mindshielded later in game is that big of a problem add sechuds to blindfolds.
> 
> > The cult stun is balanced around the fact that cult is severely weak to ranged ballistics
> 
> How on earth is one-hit stun balanced? Have you tried playing security fighting cultists? Most of the time random security officers won't have shotguns ready shift-start to blast you with and using stun you still can try to steal baton from them and win the fight that way.

usually i fight cult as a valid hungry non-security rather than security itself, nonetheless, i fight cult plenty and have lots of experience in not running into melee range of people that have stun hand ready, and not running straight after people holding bolas(which got inhand sprites within the last couple months). I usually use slips or guns(ideally shotgun) while they are running directly towards me. if they manage to get me into a corner, dodge my slip, dodge my bullets, and or otherwise have me immobile to the point that i cant escape before killing them, then i have either been outskilled or got unlucky(usually the former), as i had lost with a range advantage. either that or they snuck up on me, good on them for maintaining stealth.

shift start the cult will(usually) still be stealthed, im referring to after cult gets outed and security is armed up. that's the idea im balancing for, as it is what generally happens. stealth cult will of course be able to snag lonely unprepared officers, it is what lings and traitors usually do as well to officers running around maint alone. 

at the point where cult goes loud, running up and trying to melee an officer with a stun in your hand is a very bad idea if they have armory gear. at the moment, It is borderline impossible to kill a good security officer as a cultist without prior stealing guns yourself or landing a lucky bola. even sneaking up on them is risky and you'll have to blow 2 stuns to subdue them)or bola and then stab them to death, praying they dont have a shotgun or baton to counter you with). you'll have to EMP them as well unless you want to get outed, which has its own issues, as it fucks doors/alarms/very noticeable.

id like to note that this PR is also a massive cult nerf. several cult rounds ive played in recent memory have ended with a few crew members and i waiting idly at a barrier rune for it to drop so that we can stop the summoning, only for it to NEVER drop.  alternatively, i'm sure tons of people here can relate to hearing the cult summon announcement and a literal minute later narsie is summoned after the master summoned everyone directly onto the narsie rune tile, negating any chance for the crew to win.

with this change, cult will need to fight for their narsie summoning, final battles will 100% certainly be more entertaining unless the cult has already wiped the entire station. as someone who gets a lot of practice with cult, i can assure you that in most of my wins i see people stuck at barrier runes trying to break in while im scribing the narsie rune, and then winning via final reckoning, while most of the crew was still loyal to cult(or at the very least the crew still had a fighting force).

**Comment by imsxz** [comment]
shadowflame ruined my quadpost........

for real though i cant stress enough that this is a substantial cult nerf, just in a different aspect than the stun change. the intent of this nerf is to cripple the ability to solo win the game as a cultist. the gamemode is supposed to need teamwork to ultimately win. that is my view on cult balance.

**Comment by Archanial** [comment]
> they arent strong in melee with the change is my point.

Being stong in melee deosn't mean they need one hit stun. Add knockdown, more stamina damage, maybe even mute.

>usually i fight cult as a valid hungry non-security rather than security itself, nonetheless, i fight cult plenty and have lots of experience in not running into melee range of people that have stun hand ready, and not running straight after people holding bolas(which got inhand sprites within the last couple months).

So the stun is not even relevant here?

> I usually use slips or guns(ideally shotgun)

I used shotgun...

> shift start the cult will(usually) still be stealthed, im referring to after cult gets outed and security is armed up. that's the idea im balancing for, as it is what generally happens.

In that case good cult would likely outnumber those fighting against it and get cool armor and shields.

>stealth cult will of course be able to snag lonely unprepared officers, it is what lings and traitors usually do as well to officers running around maint alone.

That's not the same. I don't think traitors have stuns (since the carp changes) and lings still have to sting and then kill someone with armblade. 

>at the point where cult goes loud, running up and trying to melee an officer with a stun in your hand is a very bad idea if they have armory gear. at the moment...

Good? You shouldn't approach security officer with shotgun in their hands without a good plan, some teammates or a good weapon in your hand.

>with this change, cult will need to fight for their narsie summoning, final battles will 100% certainly be more entertaining unless the cult has already wiped the entire station.
Let's be honest, most of rounds either end up with massive cult dominance or just massive fails.

Rest of the changes are good but I don't see how stun should reverted because of late game.

**Comment by imsxz** [comment]
the stun isnt bullshit lategame is my point, it's necessary. in my case where you mentioned "so the stun isnt even relevant here?", it is relevant, because if a cultist lategame when im fighting cult gets past my line of defenses then I dont think to myself "wow that 1 hit stun was OP" i think "wow my aim was shit". with the current change it gives me even more chances to shoot the cultist that i was already shooting at and by all means should have already shot dead with my shotgun, a saving grace that i feel i never deserved as i lost my advantage of using a ranged weapon against a melee specialist(the cultist).

A big meme about cult that some love to spout is that you dont even need to stealth as cult - just go loud and murderbone. The current functionality of stealth further hinders cults stealth ability, making it even less viable. I have no strong opinion on whether or not stealth antags are more fun than loud antags, but as it stands, killing security as cult is basically the "go loud" button as he'll just call for help unless you additionally blow an EMP and dont get noticed.

The old stun, in cults current iteration, is 100% necessary in keeping them a threat after they get outed, assuming the crew is still capable of getting guns and stuff. Of course, some rounds the cult stealth until they get red halos, but those are exceptionally rare and need a good cult OR extraordinarily absent crew.

As for cult being flop/stomp, I can't relate to that. On bagil, where I primarily play, cults actively get shot down after getting an early game advantage, and can also come back after a few get caught early. Your posterboy powergamer cultist players won't always stomp.

**Comment by Archanial** [comment]
My nerf was aimed at early game not the late game. If you want to buff late game of the cult buff it in the other areas. Still if you need one hit ability to win a fight you're doing something wrong.

**Comment by necromanceranne** [comment]
I think it's worth pointing out that the baseball bat right now can lock you into an infinite stunloop and is a makeshift weapon, so anyone can make it. To me, this works against literally any threat and not just cultists, so I think this is an issue specifically with baseball bats being obnoxiously more powerful than they should be. They also are able to stun people by way of selfbatting too.

As for melee fighting, this is kind of the thing I've been hammering on about with how weird and obnoxiously all over the shop combat is in regards to melee combat, because power is in the wrong spots and too overwhelming. A stun hand is basically making it impossible to enter into melee with a cultist at all, since getting into melee range is an effective oneshot. This is where a cultist should be shining but it doesn't make for very fun gameplay except for the person using the spell who is just going to kill pretty much anyone who they get into a fight with. But it is needed for the cult to gather sacrifices early on. So starting to take it out as the game goes into more direct conflicts is a sound move to me and rather smart.

Yet now we have a situation where it's in the reverse and stun batons and baseball bats will do the same to the cultists if they try to enter melee against a mindshield implanted crew member. And since they got dragged into the brig, chances are they can get a baton. It's all or nothing, and we are giving too much power in a single hit or too few hits. What should be happening is that entering a melee fight with a cultist isn't just going to be a bad idea, you might struggle to meaningfully overpower them before they overpower you, and they have generally more options that contribute to their success than you. But it shouldn't be through rocket tag mechanics.

These are two problems that are two sides of the same coin and neither situation should exist where neither side ever wants to enter into this situation. Or well, cultists shouldn't be feeling like they can never enter into melee at the very least, since that is where they should be shining and they should feel confident that if they try to do so they can at least get into their comfortable spot.

Let's not even get into shotguns at the moment since we have had a lot of discussions about those across multiple prs, but there is a tenuous issue with ballistic weapons for sec in the form of shotguns, and as of right now a lot of the game has been balanced around the fact that those exist and they existed during the taser period, so much of the game is still balanced around tasers being hard countered and shotguns being a step down alternative. Cultists, after all, are immune to energy guns entirely with a mirror shield yet vulnerable to ballistics, but the mirror shield was meant to ideally counter tasers, and only actually just mandates the existence of shotguns since without shotguns there would be no ready option to fight a cultist. (Also emp spam.)

Besides, what synergies exist with the stamina damage for cultists? I think almost all their attacks are brute based or hard stuns. I think it wasn't just that it was a bad nerf per say but not actually beneficial for the cultists in any way. It also saps blood, but none of that actually is added to the cultists blood rites pool. Which would be huge given blood rites can be used to heal a cultist, among other uses.

Rather than reverting the stun removal, try to find what would be appropriate for a cultist that would put anyone who is hit with this in a clear and distinct backfoot in melee. Hell, even if that is actually buffing some of their potency elsewhere, and maybe think whether baseball bats should be a good answer to a man with a sword, as absurdist and funny as it might be.

**Comment by necromanceranne** [comment]
TL;DR Wild ass suggestion, what if it did 60 brute to the fucking face and gave the cultist the stolen blood for blood rites.

Stun hand to CAST: PAWNCH!

**Comment by BeeSting12** [comment]
Ah yes bring back the struggle of trying to get 9 people to stand on a rune. 👍

**Comment by Ziiro** [comment]
>If not seeing who's mindshielded later in game is that big of a problem add sechuds to blindfolds.

ideas guy moment: when cultists get their red eyes, give them inherent sec huds.

**Comment by Tlaltecuhtli** [comment]
make mindshield give a shorter stun, there is literal no advantage by having mindshield (only thing you get is sitting on a shard once cult steamrolls)

**Comment by Ikoden** [comment]
Buff cult stun but not to prenerf state.

**Comment by wesoda25** [comment]
just make cult stun less effective as cult moves up in pop. So at roundstart its normally effective, at red eyes a bit less, and halos a bit less.

**Comment by necromanceranne** [comment]
You wouldn't believe what I discovered. So I was thinking of opening an alternative pr (yes, I know pr jousting is bad), but while I was testing some changes, wouldn't you know it, I discovered nothing stops you scribing two stun spells onto yourself, equipping BOTH in each hand, and then hitting them one after the other with the spell.

It's literally a 1-2 stamcrit. Was everyone using it one at a time?

**Comment by AdamElTablawy** [comment]
@necromanceranne That still requires you to swap hands after the hit, which can be tricky.

**Comment by ATH1909** [comment]
Increase the integrity of barrier runes, add secHUDvision to zealot blindfolds (or, hell, even just give secHUDvision to all cultists, no matter what they're wearing), and don't undo the cult stun nerf, and I think this will be a really good rebalancing for blood cults.

**Comment by imsxz** [comment]
zealot blindfolds dont need more benefits. theres already zero reason to not get one as soon as you can as a cultist unless you arent wearing a mask. Flash protection, night vision, and medhud all rolled into one with the downside being that it's cult clothing and you might get lynched for it.

I do like wesoda's idea  a lot, it's something i've had in mind for a while without much motivation to code. If I were to do it, it would be part of a larger scale rework that would generally have the cult dissipate power as it gets more people or something to that effect, making mass conversion not the default strategy. Ideally keeping 9 or less cultists would have you at your strongest individual power(most certainly scaling with pop) but that is not in the scope of this PR.

I have extensive experience playing as and against cult, as well as watching people on both sides a ton while observing. I firmly believe that the stun was fine before the nerf. A necessary evil to make playing as the antag not feel very oppressed or incapable of fighting, like hivemind struggled with for some time, where most were very shy about fighting because they werent confident in their abilities.

I believe my changes will have a much bigger impact than the stun nerf in terms of the ability for cult players to solo carry, as well as making cheap wins a lot lot more difficult for cult to acquire.  I'm almost certain that cult wins are primarily with the cult master using final reckoning to get everyone onto the rune before the crew is able to get to the summon location. Of course winning or losing doesn't really matter, but I believe firmly that people will enjoy the gamemode a lot more if there's a climactic final fight at narsie summonings rather than a bunch of helpless crew staring in through the cults invisible&invincible barrier runes until narsie is summoned.

tl;dr stun change bad, my change good

**Comment by imsxz** [comment]
also no i dont plan on raising the barrier integrity until i see it in play. if you want a sturdy barrier, construct a cult wall or use an artificer to place one. my view on barriers is that they're supposed to buy you a few extra seconds in the event someone stumbles near whatever you're barrier-ing.

forgot to mention, this will also largely lower the effectiveness of boil blood spam, as cultists wont be able to do it safely behind barriers more than a couple times before an angry mob beats it down, potentially stopping a boil blood entirely if nobody is prepared to invoke it.

File: code/modules/antagonists/cult/blood_magic.dm [hunk 1]
```diff
@@ -436,28 +436,18 @@
 									   "<span class='userdanger'>A feeling of warmth washes over you, rays of holy light surround your body and protect you from the flash of light!</span>")
 
 		else
-			if(HAS_TRAIT(target, TRAIT_MINDSHIELD))
+			to_chat(user, "<span class='cultitalic'>In a brilliant flash of red, [L] falls to the ground!</span>")
+			L.Paralyze(160)
+			L.flash_act(1,1)
```
Comment (ExcessiveUseOfCobblestone) [inline code/modules/antagonists/cult/blood_magic.dm:441]:
```suggestion
			L.flash_act(1,TRUE)
```

File: code/game/machinery/shieldgen.dm [hunk 2]
```diff
@@ -43,26 +43,55 @@
...
+/obj/structure/emergency_shield/cult/barrier/attack_animal(mob/living/simple_animal/user)
+	if(iscultist(user))
+		parent_rune.attack_animal(user)
+	else
+		..()
+
+/obj/structure/emergency_shield/cult/barrier/Destroy()
+	if(parent_rune)
+		parent_rune.visible_message("<span class='danger'>The [parent_rune] fades away as [src] is destroyed!</span>")
+		qdel(parent_rune)
```
Comment (ExcessiveUseOfCobblestone) [inline code/game/machinery/shieldgen.dm:86]: QDELL_NULL(parent_rune)

File: code/game/machinery/shieldgen.dm [hunk 3]
```diff
@@ -43,26 +43,55 @@
...
 	desc = "A weak shield summoned by cultists to protect them while they carry out delicate rituals."
 	color = "#FF0000"
 	max_integrity = 20
 	mouse_opacity = MOUSE_OPACITY_TRANSPARENT
 	layer = ABOVE_MOB_LAYER
 
-/obj/structure/emergency_shield/invoker/emp_act(severity)
-	return
+/obj/structure/emergency_shield/cult/barrier
+	var/obj/effect/rune/parent_rune // for deleting the barrier rune that created the barrier
```
Comment (ExcessiveUseOfCobblestone) [inline code/game/machinery/shieldgen.dm:70]: new variables go below pre-existing ones, and include documentation

File: code/modules/antagonists/cult/blood_magic.dm [hunk 4]
```diff
@@ -436,28 +436,18 @@
 									   "<span class='userdanger'>A feeling of warmth washes over you, rays of holy light surround your body and protect you from the flash of light!</span>")
 
 		else
-			if(HAS_TRAIT(target, TRAIT_MINDSHIELD))
+			to_chat(user, "<span class='cultitalic'>In a brilliant flash of red, [L] falls to the ground!</span>")
+			L.Paralyze(160)
```
Comment (ExcessiveUseOfCobblestone) [inline code/modules/antagonists/cult/blood_magic.dm:440]: time defines

File: code/modules/antagonists/cult/blood_magic.dm [hunk 5]
```diff
@@ -436,28 +436,18 @@
...
-				else if(iscarbon(target))
-					var/mob/living/carbon/C = L
-					C.silent += 6
-					C.stuttering += 15
-					C.cultslurring += 15
-					C.Jitter(15)
+				C.silent += 6
+				C.stuttering += 15
+				C.cultslurring += 15
+				C.Jitter(15)
```
Comment (ExcessiveUseOfCobblestone) [inline code/modules/antagonists/cult/blood_magic.dm:450]: time defines

File: code/game/machinery/shieldgen.dm [hunk 6]
```diff
@@ -43,26 +43,55 @@
...
+	else
+		..()
+
+/obj/structure/emergency_shield/cult/barrier/Destroy()
+	if(parent_rune)
+		parent_rune.visible_message("<span class='danger'>The [parent_rune] fades away as [src] is destroyed!</span>")
+		qdel(parent_rune)
+	..()
+
+/obj/structure/emergency_shield/cult/barrier/proc/Toggle()
```
Comment (ExcessiveUseOfCobblestone) [inline code/game/machinery/shieldgen.dm:89]: doc

File: code/modules/antagonists/cult/blood_magic.dm [hunk 7]
```diff
@@ -436,28 +436,18 @@
...
```
Comment (ExcessiveUseOfCobblestone) [inline code/modules/antagonists/cult/blood_magic.dm:450]:
```suggestion
				C.Jitter(1.5 SECONDS)
```

File: code/modules/antagonists/cult/blood_magic.dm [hunk 8]
```diff
@@ -436,28 +436,18 @@
...
-					var/mob/living/silicon/S = L
-					S.emp_act(EMP_HEAVY)
-				else if(iscarbon(target))
-					var/mob/living/carbon/C = L
-					C.silent += 6
-					C.stuttering += 15
-					C.cultslurring += 15
-					C.Jitter(15)
+				C.silent += 6
+				C.stuttering += 15
```
Comment (ExcessiveUseOfCobblestone) [inline code/modules/antagonists/cult/blood_magic.dm:448]:
```suggestion
				C.stuttering += (1.5 SECONDS)
```

File: code/modules/antagonists/cult/blood_magic.dm [hunk 9]
```diff
@@ -436,28 +436,18 @@
...
+				C.cultslurring += 15
```
Comment (ExcessiveUseOfCobblestone) [inline code/modules/antagonists/cult/blood_magic.dm:449]:
```suggestion
				C.cultslurring += (1.5 SECONDS)
```

File: code/modules/antagonists/cult/blood_magic.dm (hunk 7 above)
Comment (imsxz) [reply code/modules/antagonists/cult/blood_magic.dm:450]: shu tup

**Comment by imsxz** [comment]
why does random shit use different seconds it makes no sense what the fuck

**Comment by imsxz** [comment]
jitter 15 looked normal ok?

File: code/modules/antagonists/cult/blood_magic.dm (hunk 9 above)
Comment (imsxz) [reply code/modules/antagonists/cult/blood_magic.dm:449]: that is so misleading tho because its 15 seconds and not 1.5

**Comment by Tlaltecuhtli** [comment]
if you are changing cult can you remove the shitty heal from conversion? the meta of critting people is cancer and a dumb shortcut to something you have all tools to do it without

**Comment by imsxz** [comment]
if i did that then everyone would cry when they get converted after a jug/longsword crits them and that will make ppl hate the gamemode even more

**Comment by imsxz** [comment]
god i love cobby

**Comment by zeskorion** [comment]
combat stuns gay

monkey is the only balanced conversion antag

**Comment by ExcessiveUseOfCobblestone** [comment]
we can look at btfoing combat stuns again once I see how the barrier/mass summon cheese affects cult 😏

**Comment by saprasam** [comment]
just bring back clock cult

**Pull request merged**